internal/app/api/payments.go

```diff
@@ -121,60 +138,71 @@ func ListPaymentsHandler(db *mongo.Database) http.HandlerFunc {
 		})
 		if err != nil {
 			handleServerError(w, r, err)
+
 			return
 		}
 	}
 }
 
-func ReadPaymentHandler(db *mongo.Database) http.HandlerFunc {
+func readPaymentHandler(db *mongo.Database) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
+		paymentID := mux.Vars(r)["paymentID"]
 
-		paymentId := mux.Vars(r)["paymentId"]
-
-		identifier, err := payments.IdentifierFromString(paymentId)
+		identifier, err := payments.IdentifierFromString(paymentID)
 		if err != nil {
 			w.WriteHeader(http.StatusNotFound)
+
 			return
 		}
 
 		ret := db.Collection(payments.Collection).FindOne(r.Context(), identifier)
 		if ret.Err() != nil {
-			if ret.Err() == mongo.ErrNoDocuments {
+			if errors.Is(ret.Err(), mongo.ErrNoDocuments) {
 				w.WriteHeader(http.StatusNotFound)
+
 				return
 			}
+
 			handleServerError(w, r, ret.Err())
+
 			return
 		}
-		ob := &payments.Payment{}
-		err = ret.Decode(ob)
+
+		payment := &payments.Payment{}
+
+		err = ret.Decode(payment)
 		if err != nil {
 			handleServerError(w, r, err)
+
 			return
 		}
 
 		err = json.NewEncoder(w).Encode(sharedapi.BaseResponse[payments.Payment]{
-			Data: ob,
+			Data: payment,
 		})
 		if err != nil {
 			handleServerError(w, r, err)
+
 			return
 		}
 	}
 }
 
-func PaymentsRouter(
+func paymentsRouter(
 	db *mongo.Database,
-	useScopes bool) *mux.Router {
+	useScopes bool,
+) *mux.Router {
 	router := mux.NewRouter()
 	router.Use(func(h http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			w.Header().Set("Content-Type", "application/json")
 			h.ServeHTTP(w, r)
 		})
 	})
-	router.Path("/payments").Methods(http.MethodGet).Handler(WrapHandler(useScopes, ListPaymentsHandler(db), ScopeReadPayments, ScopeWritePayments))
-	router.Path("/payments/{paymentId}").Methods(http.MethodGet).Handler(WrapHandler(useScopes, ReadPaymentHandler(db), ScopeReadPayments, ScopeWritePayments))
+	router.Path("/payments").Methods(http.MethodGet).
+		Handler(wrapHandler(useScopes, listPaymentsHandler(db), scopeReadPayments, scopeWritePayments))
```

Comment:
I think we can remove the scopes support on this application. And all others stuff related to OAuth2.
We plan to handle access using an external gateway.
@flemzord